lib/connectors_async/job_runner.rb

```diff
@@ -8,14 +8,22 @@
 
 require 'connectors_app/config'
 require 'connectors_async/job_store'
+require 'connectors_shared/constants'
 require 'connectors_shared/job_status'
 require 'connectors_shared/exception_tracking'
 require 'connectors_shared/logger'
 
 module ConnectorsAsync
   class JobRunner
+    class JobStuckError < StandardError; end
+
     def initialize(max_threads:)
-      @pool = Concurrent::ThreadPoolExecutor.new(min_threads: 1, max_threads: max_threads, max_queue: 0)
+      @pool = Concurrent::ThreadPoolExecutor.new(
+        min_threads: 1,
+        max_threads: max_threads,
+        max_queue: 0,
+        idletime: ConnectorsShared::Constants::IDLE_SLEEP_TIME + 1 # we +1 just so that thread.sleep manages to finish by the idle timeout
```

Comment:
Let's make the unit explicit:
```suggestion
        idletime: ConnectorsShared::Constants::IDLE_SLEEP_TIME + 1.seconds # we +1 just so that thread.sleep manages to finish by the idle timeout
```